### PR TITLE
Add "Dynamic observation and interruption of sparql queries" blogpost

### PR DESCRIPTION
**Pull request by RobinTF**
/cc @joka921

### CODE REVIEW

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+---
+title: "Dynamic Observation and Interruption of SPARQL Queries"
+date: 2024-03-29T03:19:19+01:00
+author: "Robin Textor-Falconi"
+authorAvatar: "img/ada.jpg"
+tags: ["SPARQL", "WebSocket", "HTTP", "Synchronization", "Atomics", "QLever"]
+categories: ["project"]
+image: "img/project-dynamic-observation-and-interruption-of-sparql-queries/teaser.png"
+draft: true
```
Comment (joka921) [inline content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]:
```suggestion
draft: false
```
(Then we don't wonder why it is not shown when deploying the post)

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+3. [Embracing Cancel Culture](#embracing-cancel-culture)
+    1. [Cancellation via WebSockets](#cancellation-via-websockets)
+    2. [Cancellation Handles and Watchdogs](#cancellation-handles-and-watchdogs)
+    3. [Collateral Benefits](#collateral-benefits)
+4. [Conclusion](#conclusion)
+
+
+## Real-Time is more than quality of life
+
+QLever is a SPARQL-compliant knowledge-base engine that allows to blazingly fast execution of SPARQL queries on giant RDF
```
Comment (joka921) [inline content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]: The term `knowledge graph` is more common nowadays.

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+data graphs. However, even though most common queries run in less than a couple of seconds, of course, more expensive
+queries exist that will take dozens of seconds, minutes or even whole hours to compute. These kinds of queries will
+eventually be completed but generally, it's hard to tell how long a novel query will take from an outside perspective.
```
Comment (joka921) [inline content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]:
```suggestion
eventually be completed but generally, it's hard to tell how long a query will take from an outside perspective.
```

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md (hunk 1 above)
Comment (RobinTF) [reply content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]: I probably should adjust the date too before merging this so it matches the moment it gets published

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md (hunk 2 above)
Comment (joka921) [inline content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]: typo "tthat allows *the* blazingly fast execution" or "that allows to blazingly fast execute"

File: content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+When a new connection is being established and the query ID is not yet in use (usually this means the query has not started yet)
+the server just keeps the connection alive without sending anything else until a query with this specific ID starts. There can
+be multiple WebSockets waiting for the exact same query which doesn't really help keeping things simple, mainly to allow
+observations by server administrators in case queries are hogging up a lot of resources for no apparent reason. Once all
+listeners are connected to QLever and a query makes progress it has to broadcast the latest update to all listeners. Of course,
+there are some important considerations to be made here. The broadcast operation cannot be blocking in any way, otherwise,
+a slow WebSocket client can just slow down queries artificially. Also ideally all WebSockets should be able to operate
+independently of each other for maximum efficiency, but due to technical limitations of `Boost.Asio` we ended up running all
+io operations in a somewhat synchronized manner to avoid race conditions.
```
Comment (joka921) [inline content/post/project-dynamic-observation-and-interruption-of-sparql-queries.md]: I wouldn't say that this is true. Only very cheap ramping / async initiate functions are synchronized and that's the way it should be. Not need to pessimize your achievements here.

**Review by joka921 — COMMENTED** [review]
I found some small typos which you could fix.
Overall the still is pleasant to read, but sometimes the "chatty conversational style" is out of sync with the "complicated technical" contents, but that is not a serious issue but probably a question of personal style.
Just fix the small stuff and then we are good to go.

**Pull request merged**